application/analyst/services/state_updater.py

```
@@ -328,13 +349,64 @@ def _update_knowledge(
+            ending_state = ""
+            ending_emotion = ""
+            carry_over_question = ""
+            next_opening_hint = ""
+
+            chapter = None
+            if self.chapter_repository:
+                try:
+                    chapter = self.chapter_repository.get_by_novel_and_number(
+                        NovelId(novel_id), chapter_number
+                    )
+                except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)